src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.cs

```diff
@@ -0,0 +1,446 @@
+                return macOS;
+            return name;
+        }
+
+        private static DiagnosticDescriptor SelectRule(PlatformAttributeType attributeType)
+        {
+            if (attributeType == PlatformAttributeType.MinimumOSPlatformAttribute)
+                return MinimumOsRule;
+            if (attributeType == PlatformAttributeType.ObsoletedInOSPlatformAttribute)
+                return ObsoleteRule;
+            return RemovedRule;
+        }
+
+        private static AttributeData[] FindAllPlatformAttributesApplied(ImmutableArray<AttributeData> immediateAttributes, INamedTypeSymbol parent)
+        {
+            using var builder = ArrayBuilder<AttributeData>.GetInstance();
+            foreach (AttributeData attribute in immediateAttributes)
+            {
+                if (s_osPlatformAttributes.Contains(attribute.AttributeClass.Name))
+                {
+                    builder.Add(attribute);
+                }
+            }
+
+            while (parent != null)
+            {
+                var current = parent.GetAttributes();
+                foreach (var attribute in current)
+                {
+                    if (s_osPlatformAttributes.Contains(attribute.AttributeClass.Name) &&
+                        TargetPlatformAttribute != attribute.AttributeClass.Name)
+                    {
+                        builder.Add(attribute);
+                    }
+                }
+                parent = parent.BaseType;
+            }
+            return builder.ToArray();
+        }
+
+        private static bool IsSuppressedByAttribute(PlatformAttributeInfo diagnosingAttribute, ISymbol containingSymbol)
+        {
+            while (containingSymbol != null)
+            {
+                var attributes = containingSymbol.GetAttributes();
+                if (attributes != null)
+                {
+                    foreach (AttributeData attribute in attributes)
+                    {
+                        if (diagnosingAttribute.AttributeType.ToString() == attribute.AttributeClass.Name)
+                        {
+                            if (PlatformAttributeInfo.TryParsePlatformAttributeInfo(attribute, out PlatformAttributeInfo parsedAttribute))
+                            {
+                                if (IsOSPlatformsEqual(diagnosingAttribute.PlatformName, parsedAttribute.PlatformName) &&
+                                    AttributeVersionsMatch(diagnosingAttribute, parsedAttribute))
+                                {
+                                    return true;
+                                }
+                            }
+                        }
+                    }
+                }
+                containingSymbol = containingSymbol.ContainingSymbol;
+            }
+            return false;
+        }
+
+        private static bool AttributeVersionsMatch(PlatformAttributeInfo diagnosingAttribute, PlatformAttributeInfo osAttribute)
```

Comment:
It seems this method can be inlined and the below can be made a static local function of the caller.

Reply:
>  below can be made a static local function of the caller.

Below method has other callers